pilot/pkg/model/generic_xds_cache.go

```diff
@@ -0,0 +1,264 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/config/schema/kind"
+	"istio.io/istio/pkg/util/sets"
+)
+
+type XdsCacheImpl struct {
+	cds xdsCache[uint64]
+	eds xdsCache[uint64]
+	rds xdsCache[uint64]
+	sds xdsCache[string]
+}
+
+// GenericXdsCache interface defines a store for caching XDS responses.
+// All operations are thread safe.
+type GenericXdsCache interface {
```

Comment:
keeping the name of this as "XdsCache" and renaming the interface in xds_cache as "typedXdsCache" may be more readable as that is per type?

Reply:
SG, i added `GenericXdsCache` first before renaming the original XdsCache